cpp/include/cudf/io/parquet.hpp

```diff
@@ -489,6 +499,17 @@ class parquet_writer_options {
    */
   auto get_row_group_size_rows() const { return _row_group_size_rows; }
 
+  /**
+   * @brief Sets partitions.
+   *
+   * @param partitions Partitions of input table in {start_row, num_rows} pairs. If specified, must
+   * be same size as number of sinks in sink_info
+   */
```

Comment:
This has me wondering if the interface should be setup such that you have to specify the sinks and partitions all in one place, to centralize all error checking one spot.

Reply:
Not only that, the bigger smell is that `table_input_metadata::user_data` is now a vector with per-file map. And that is not even declared in this header. Can't seems to figure out how to bring these together.

Reply:
Here's something I can do: Bring the `user_metadata` out of `table_input_metadata` and rename the latter to `table_input_schema`. `user_data` can go directly in the `parquet_writer_options`.

Reply:
sounds good, worth giving it a try (and see how the tests are affected).